### PR TITLE
feat!: IgnoreError Recursively for tables and columns

### PR DESCRIPTION
**Pull request by yevgenypats**
🎉 Thank you for making CloudQuery awesome by submitting a PR 🎉

#### Summary

This adds recursive logic for both columns and tables so we dont have to put it in every column and relation. 

This need more tests but ready for initial review.

Closes https://github.com/cloudquery/cq-provider-sdk/pull/323

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/.github/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: provider/schema/table.go
Comment (erezrokah) [inline provider/schema/table.go:102]: What is the use case for running the table ignore error function for specific columns?
Are we expecting similar errors to happen?

When I added an ignore error for columns it was super specific to the data type of the column and the provider API:
https://github.com/cloudquery/cq-provider-gcp/blob/49238085984bc25468ced7484761a3ceb1b7a99e/client/error.go#L59

I'd expect that table level errors are more about failures to fetch the data, and column level errors are more about failures to parse/normalize the data

**Review by erezrokah — COMMENTED** [review]
I'll let others reviews comment too, but WDYT about starting with the table level changes, and then moving to the column level changes?

File: provider/schema/table.go
Comment (yevgenypats) [reply provider/schema/table.go:102]: Where do we use ColumnIgnoreError apart from this case? trying to find it with regex...

Comment (erezrokah) [reply provider/schema/table.go:102]: I think I might have been the first consumer of that feature as I had to do another fix in our testing logic to get it to work https://github.com/cloudquery/cq-provider-sdk/pull/283

Comment (yevgenypats) [reply provider/schema/table.go:102]: I see. so seems to me it will be best to remove this feature as well as it is unused, so less documentation, less bugs. and given this PR was the first consumer I suggest to implement it by wrapping up `IPNetResolver` with `GCPIPNetResolver` that will also ignore this error. WDYT ?

Comment (bbernays) [reply provider/schema/table.go:102]: So why not just only recursively propagate the table level error handler?

Comment (erezrokah) [reply provider/schema/table.go:102]: Ah well, that's how I started and the realized we have in the SDK. I'll open a new issue for that.

Comment (yevgenypats) [reply provider/schema/table.go:102]: Or even better add to the `IPNetResolver` a check that if it's empty just don't do anything. seems to me like a good default behaviour for all cases or just set it to null instead. both seems good to me.

Comment (erezrokah) [reply provider/schema/table.go:102]: Opened https://github.com/cloudquery/cq-provider-sdk/issues/327 and https://github.com/cloudquery/cq-provider-sdk/issues/326.

Didn't want to change the default `IPNetResolver` back then as I only had a very specific use case for the GCP provider. Didn't see empty strings coming from other APIs.

Comment (yevgenypats) [reply provider/schema/table.go]: I actually already pushed it as it's less code in this PR, so I just need to add a test now

File: provider/execution/execution.go
Comment (disq) [inline provider/execution/execution.go:95]: You don't need this since you have line 96?

**Review by erezrokah — COMMENTED** [review]
Add two more comments

File: provider/execution/execution.go
Comment (yevgenypats) [reply provider/execution/execution.go:95]: I think I do because parent is not set in table automatically (i.e by the user). I can also move this function to execution and not to table if that will make more sense.

File: provider/execution/execution.go
Comment (yevgenypats) [reply provider/execution/execution.go:417]: Good point. Yeah we can do that. It might be needed in the future but Im all in on removing now and adding in the future if needed.

File: provider/execution/execution.go
Comment (yevgenypats) [reply provider/execution/execution.go:240]: I think there is already a debug log there on line 241

**Review by erezrokah — APPROVED** [review]
🚀  @yevgenypats please squash merge with a commit title prefix of `feat!: ` as this is a breaking change

**Comment by yevgenypats** [comment]
@erezrokah will do! Will wait on @disq approval (regarding comment) and will squash merge 🚀

File: provider/execution/execution.go
Comment (disq) [reply provider/execution/execution.go:95]: But this is overridden by L96, so it has no effect?

**Review by disq — APPROVED** [review]
LGTM

**Pull request merged**
